test/e2e_env/universal/timeout/meshtimeout.go

```diff
@@ -0,0 +1,99 @@
+package timeout
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/test/e2e_env/universal/env"
+	. "github.com/kumahq/kuma/test/framework"
+)
+
+func PluginTest() {
+	meshName := "meshtimeout"
+	faultInjection := fmt.Sprintf(`
+type: FaultInjection
+mesh: "%s"
+name: fi1
+sources:
+   - match:
+       kuma.io/service: demo-client
+destinations:
+   - match:
+       kuma.io/service: test-server
+       kuma.io/protocol: http
+conf:
+   delay:
+     percentage: 100
+     value: 5s
+`, meshName)
+
+	timeout := fmt.Sprintf(`
+type: MeshTimeout
+name: default
+mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: test-server
+      default:
+        connectionTimeout: 20s
+        http:
+          requestTimeout: 2s
+`, meshName)
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MeshUniversal(meshName)).
+			Install(YamlUniversal(faultInjection)).
+			Install(DemoClientUniversal("demo-client", meshName,
+				WithTransparentProxy(true)),
+			).
+			Install(TestServerUniversal("test-server", meshName,
+				WithArgs([]string{"echo", "--instance", "universal-1"})),
+			).
+			Setup(env.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	E2EAfterAll(func() {
+		Expect(env.Cluster.DeleteMeshApps(meshName)).To(Succeed())
+		Expect(env.Cluster.DeleteMesh(meshName)).To(Succeed())
+	})
+
+	It("should reset the connection by timeout", func() {
+		By("Checking requests succeed")
+		Eventually(func(g Gomega) {
+			stdout, _, err := env.Cluster.Exec("", "", "demo-client",
+				"curl", "-v", "--fail", "test-server.mesh")
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(stdout).To(ContainSubstring("HTTP/1.1 200 OK"))
+		}).Should(Succeed())
+
+		By("check requests take over 5s")
+		start := time.Now()
+		_, _, err := env.Cluster.Exec("", "", "demo-client",
+			"curl", "-v", "--fail", "test-server.mesh")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(time.Since(start)).To(BeNumerically(">", time.Second*5))
+
+		By("apply a new policy")
+		Expect(env.Cluster.Install(YamlUniversal(timeout))).To(Succeed())
+
+		By("eventually requests timeout consistently")
+		Eventually(func() string {
+			stdout, _, _ := env.Cluster.Exec("", "", "demo-client",
+				"curl", "-v", "test-server.mesh")
+			return stdout
+		}).Should(ContainSubstring("upstream request timeout"))
+		Consistently(func() string {
+			stdout, _, _ := env.Cluster.Exec("", "", "demo-client",
+				"curl", "-v", "test-server.mesh")
+			return stdout
+		}).Should(ContainSubstring("upstream request timeout"))
```

Comment:
I'm looking at the defaults durations and intervals
```
func TestE2E(t *testing.T) {
	SetDefaultConsistentlyDuration(time.Second * 5)
	SetDefaultConsistentlyPollingInterval(time.Millisecond * 200)
	SetDefaultEventuallyPollingInterval(time.Millisecond * 500)
	SetDefaultEventuallyTimeout(time.Second * 10)
	test.RunSpecs(t, "E2E Universal Suite")
}
```
and I feel like it might be too tight for this test. Consistently does not make sense in such case, no?

Reply:
changed